aws/rust-runtime/aws-config/src/lib.rs

```diff
@@ -372,8 +442,8 @@ mod loader {
         ///
         /// Update the `ProviderConfig` used for all nested loaders. This can be used to override
         /// the HTTPs connector used by providers or to stub in an in memory `Env` or `Fs` for testing.
-        /// This **does not set** the HTTP connector used when sending operations. To change that
-        /// connector, use [ConfigLoader::http_connector].
+        ///
+        /// NOTE: Update the provider config with care because it will override any other settings.
```

Comment:
Can we be more specific about what this will override? I'm wondering if we should rename `ProviderConfig` to better highlight the stuff it provides. I'm thinking of something like this:

- `OsProvider`
- `PlatformProvider`
- `SystemProvider`